addons/l10n_in/models/account_move_line.py

```diff
@@ -38,3 +66,156 @@ def _l10n_in_check_invalid_hsn_code(self):
                 product_line=self.product_id.name or self.name
             )
         return False
+
+    def get_l10n_in_tax_tag_ids(self):
+
+        def get_tag_ids(*refs):
+            return [self.env.ref(ref).id for ref in refs]
+
+        return {
+            'gst_rc': get_tag_ids(
+                'l10n_in.tax_tag_base_sgst_rc', 'l10n_in.tax_tag_sgst_rc',
+                'l10n_in.tax_tag_base_cgst_rc', 'l10n_in.tax_tag_cgst_rc',
+                'l10n_in.tax_tag_base_igst_rc', 'l10n_in.tax_tag_igst_rc',
+                'l10n_in.tax_tag_base_cess_rc', 'l10n_in.tax_tag_cess_rc'
+            ),
+            'gst': get_tag_ids(
+                'l10n_in.tax_tag_base_sgst', 'l10n_in.tax_tag_sgst',
+                'l10n_in.tax_tag_base_cgst', 'l10n_in.tax_tag_cgst',
+                'l10n_in.tax_tag_base_igst', 'l10n_in.tax_tag_igst',
+                'l10n_in.tax_tag_base_cess', 'l10n_in.tax_tag_cess'
+            ),
+            'nil': get_tag_ids(
+                'l10n_in.tax_tag_exempt', 'l10n_in.tax_tag_nil_rated', 'l10n_in.tax_tag_non_gst_supplies'
+            ),
+            'export': get_tag_ids(
+                'l10n_in.tax_tag_zero_rated', 'l10n_in.tax_tag_base_igst', 'l10n_in.tax_tag_igst',
+                'l10n_in.tax_tag_base_cess', 'l10n_in.tax_tag_cess'
+            ),
+            'igst_lut': get_tag_ids('l10n_in.tax_tag_base_igst_lut')
+        }
+
+    @api.depends('move_id.l10n_in_gst_treatment', 'move_id.amount_total', 'move_id.l10n_in_state_id', 'company_id', 'tax_tag_ids')
+    def _compute_l10n_in_gstr_section(self):
+
+        def has_tags(category, tax_tags):
+            return any(tag in tax_tags for tag in tax_tags_ids[category])
+
+        def is_invoice(move):
+            return move.is_inbound() and not move.debit_origin_id
+
+        def get_transaction_type(move):
+            return 'intra_state' if move.l10n_in_state_id == move.company_id.state_id else 'inter_state'
+
+        def get_section(line):
+            move = line.move_id
+            gst_treatment = move.l10n_in_gst_treatment
+            transaction_type = get_transaction_type(move)
+            tags = line.tax_tag_ids.ids
+            is_inv = is_invoice(move)
+            amt_limit = 100000 if line.invoice_date >= date(2024, 11, 1) else 250000
+
+            # If no relevant tags are found, or the tags do not match any category, mark as out of scope
+            if not tags or not any(has_tags(c, tags) for c in tax_tags_ids):
+                return 'sale_out_of_scope'
+
+            # B2CS: Unregistered or Consumer sales with gst tags
+            if gst_treatment in ('unregistered', 'consumer') and has_tags('gst', tags):
+                if transaction_type == 'intra_state':
+                    return 'sale_b2cs'
+                if (
+                    transaction_type == 'inter_state' and
+                    (
+                        (is_inv and move.amount_total <= amt_limit) or
```

Comment:
```suggestion
                        is_inv and move.amount_total <= amt_limit or
```
No need for () as it is all or